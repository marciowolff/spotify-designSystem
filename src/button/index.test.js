
import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import Button from './index'

const noop = jest.fn()

it('Button should be a function', () => {
  expect(typeof Button).toBe('function')
})

it('Should Button to match snapshot', () => {
  const tree = renderer.create(<Button handleClick={noop} />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('Should Button contain text', () => {
  const wrapper = shallow(<Button handleClick={noop}>botão</Button>)

  expect(wrapper.text()).toBe('botão')
})

it('Should Button disabled', () => {
  const wrapper = shallow(<Button disabled handleClick={noop}>botão</Button>)

  expect(wrapper.is('[disabled]')).toBeTruthy()
})

it('Should Button add className', () => {
  const wrapper = shallow(<Button className="btnTeste">botão</Button>)

  expect(wrapper.hasClass('btnTeste')).toBeTruthy()
})

it('Should Button type link', () => {
  const wrapper = shallow(<Button type="link">botão</Button>)

  expect(wrapper.find('a')).toBeTruthy()
})

it('Should Button click', () => {
  const wrapper = shallow(<Button handleClick={noop}>botão</Button>)
  wrapper.find('button').simulate('click')

  expect(noop).toHaveBeenCalledTimes(1)
})
