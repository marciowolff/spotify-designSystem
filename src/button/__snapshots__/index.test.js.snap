// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Button to match snapshot 1`] = `
<button
  className="button "
  onClick={[MockFunction]}
  type="button"
/>
`;
